server/src/main/java/io/druid/server/coordination/DruidServerMetadata.java

```diff
@@ -40,7 +43,10 @@ public DruidServerMetadata(
       @JsonProperty("maxSize") long maxSize,
       @JsonProperty("type") String type,
       @JsonProperty("tier") String tier,
-      @JsonProperty("priority") int priority
+      @JsonProperty("priority") int priority,
```

Comment:
I see your point. I think it's possible to have DruidServerMetadata contain a minimum set of metadata, and let different Druid nodes extend differently to have more metadata information that is specific to the node. A follow up PR would be fine I think.